src/mono/wasm/runtime/loader/libraryInitializers.ts

```diff
@@ -0,0 +1,40 @@
+import { MonoConfig, RuntimeAPI } from "../types";
+import { BootJsonData } from "../types/blazor";
+import { appendUniqueQuery, toAbsoluteBaseUri } from "./assets";
+
+export async function fetchInitializers(moduleConfig: MonoConfig, bootConfig: BootJsonData): Promise<void> {
+    const libraryInitializers = bootConfig.resources.libraryInitializers;
+    if (!libraryInitializers) {
+        return;
+    }
+
+    if (!moduleConfig.libraryInitializers) {
+        moduleConfig.libraryInitializers = [];
+    }
+
+    const initializerFiles = Object.keys(libraryInitializers);
+    await Promise.all(initializerFiles.map(f => importInitializer(f)));
+
+    async function importInitializer(path: string): Promise<void> {
+        const adjustedPath = appendUniqueQuery(toAbsoluteBaseUri(path));
+        const initializer = await import(/* webpackIgnore: true */ adjustedPath);
```

Comment:
@pavelsavara Should I transform these es6 modules as a new type of asset and instantiate them as part of `assets.ts` download machinery?

Reply:
I think that since this is 3rd party modules, timing is key to answering this question.
- Those downloads could start after DLLs. 
- The modules do not block start of the runtime.
- They should probably instantiate only after runtime (and Blazor) is already running.

You could possibly pre-fetch them via DOM element (and hash).
The timing of when the dynamic `import` would instantiate the script is not really easy to control, right ?

So overall, I thing that they belong to the list of assets, but they are instantiated much later, not in the loader.

Reply:
Thanks! That helped me think more broadly.

Actualy, blazor does use this mechanism to allow users to create "single file bundle" https://devblogs.microsoft.com/dotnet/custom-deployment-layout-for-blazor-webassembly-apps/. The modules must be ready before `onConfigLoaded` blazor callback is called. So I will move them right after config download and add our own `onConfigLoaded` supported callback.

In future we can add MSBuild metadata, split them and download them in various parts of the livecycle.